test/spec/components/alerts.component.spec.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+describe("Component: Alerts", function(){
+
+  var $scope, $compile, element;
+  beforeEach(inject(function(_$compile_, _$rootScope_) {
+      $scope = _$rootScope_;
+      $compile = _$compile_;
+      $scope.alerts = {
+        danger: {
+          type: "error",
+          message: "An error happened and stuff.",
+          links: [
+            {
+              label: "test_link"
+            }
+          ]
+        },
+        warning: {
+          type: "warning",
+          message: "I'm going to count to five."
+        },
+        success: {
+          type: "success",
+          message: "Huzzah!"
+        },
+        info: {
+          message: "This is informational."
+        }
+      };
+
+      // Create an instance of the component
+      element = angular.element('<alerts alerts="alerts"></alerts>');
+      element = $compile(element)($scope); // Compile the component
+      $scope.$digest(); // Update the HTML
+  }));
+
+  // Ensure four alert elements exist
+  it("contain alert elements", function () {
+    let allAlerts = element.find('.alert');
+    expect(allAlerts.length).toBe(4); // four alerts exist
+  });
+
+  // Ensure danger alert exists and is correct
+  it("show danger alert", function(){
+    let danger = $(element.find('.alert')[0]);
+    expect(danger.hasClass('alert-danger')).toBeTruthy(); // danger alert has correct class
+    expect(danger.text()).toContain("An error happened and stuff."); // danger alert contains correct text
+  });
+
+  // Ensure warning alert exists and is correct
+  it("show warning alert", function(){
+    let warning = $(element.find('.alert')[1]);
+    expect(warning.hasClass('alert-warning')).toBeTruthy(); // warning alert has correct class
+    expect(warning.text()).toContain("I'm going to count to five."); // warning alert contains correct text
+  });
+
+  // Ensure success alert exists and is correct
+  it("show success alert", function(){
+    let success = $(element.find('.alert')[2]);
+    expect(success.hasClass('alert-success')).toBeTruthy(); // success alert has correct class
+    expect(success.text()).toContain("Huzzah!"); // success alert contains correct text
+  });
+
+  // Ensure info alert exists and is correct
+  it("show info alert", function(){
+    let info = $(element.find('.alert')[3]);
+    expect(info.hasClass('alert-info')).toBeTruthy(); // info alert has correct class
+    expect(info.text()).toContain("This is informational."); // info alert contains correct text
+  });
+
+  // Ensure `close` function is called when close button is clicked, and the
+  // hidden state is true.
+  it("call contoller close function", function(){
+    $scope.onCloseSpy = jasmine.createSpy('onCloseSpy');
```

Comment:
+1 to using a `spy`, I think this may be the first in our tests?

Reply:
Just noticed a typo there too... "call contoller close function"